docs/reference/packages.rst

```diff
@@ -1,314 +1,150 @@
+    >>> mip.install("github:org/user/path/package.json")
+
+If no json file is specified, then "package.json" is implicitly added::
+
+    >>> mip.install("http://example.com/x/")
+    >>> mip.install("github:org/repo")
+    >>> mip.install("github:org/repo", version="branch-or-tag")
+
+
+Using ``mip`` on the Unix port
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On the Unix port, ``mip`` can be used at the REPL as above, and also by using ``-m``::
+
+    $ ./micropython -m mip install pkgname-or-url
+    $ ./micropython -m mip install pkgname-or-url@version
+
+The ``--install-target=path`` and ``--no-install-mpy`` arguments can be set::
+
+    $ ./micropython -m mip install --install-target=third-party pkgname
+    $ ./micropython -m mip install --install-target=third-party-source --no-install-mpy pkgname
+
+Installing packages with ``mpremote``
+-------------------------------------
+
+The :term:`mpremote` tool also includes the same functionality as ``mip`` and can
+be used to install packages to a connected device::
+
+    $ mpremote install pkgname
+    $ mpremote install pkgname@x.y
+    $ mpremote install http://example.com/x/y/foo.py
+    $ mpremote install github:org/repo
+    $ mpremote install github:org/repo@branch-or-tag
+
+The ``--install-target=path`` and ``--no-install-mpy`` arguments can be set::
+
+    $ mpremote install --target=/flash/third-party pkgname
+    $ mpremote install --no-mpy pkgname
+
+Installing packages manually
+----------------------------
+
+Packages can also be installed (in either .py or .mpy form) by manually copying
+the files to the device. Depending on the board this might be via USB Mass Storage,
+the :term:`mpremote` tool (e.g. ``mpremote fs cp path/to/package.py :package.py``),
+:term:`webrepl`, etc.
+
+Writing & publishing packages
+-----------------------------
+
+Publishing to :term:`micropython-lib` is the easiest way to make your package
+broadly accessible to MicroPython users, and automatically available via
+``mip`` and ``mpremote`` and compiled to bytecode. See
+https://github.com/micropython/micropython-lib for more information.
+
+To write a "self-hosted" package that can be downloaded by ``mip`` or
+``mpremote``, you need a static webserver (or GitHub) to host either a
+single .py file, or a package.json file alongside your .py files.
+
+A typical package.json for an example ``mlx90640`` library looks like::
+
+    {
+      "urls": [
+        ["mlx90640/__init__.py", "github:org/micropython-mlx90640/mlx90640/__init__.py"],
+        ["mlx90640/utils.py", "github:org/micropython-mlx90640/mlx90640/utils.py"]
+      ],
+      "deps": [
+        ["collections-defaultdict", "latest"],
+        ["os-path", "latest"]
+      ],
+      "version": "0.2"
+    }
+
+This includes two files, hosted at a GitHub repo named
+``org/micropython-mlx90640``, which install into the ``mlx90640`` directory on
+the device. It depends on ``collections-defaultdict`` and ``os-path`` which will
+be installed automatically.
```

Comment:
is it worth adding some docs about creating and using your own index?  maybe in a new section?

Reply:
Separate PR :)

Reply:
OK!